Dungeoneer/src/com/interrupt/dungeoneer/entities/Monster.java

```diff
@@ -413,6 +417,10 @@ public boolean doPainRoll(int damage) {
 		return Game.rand.nextFloat() <= (painChance + (damageMod * 0.5f));
 	}
 
+    protected boolean shouldFlee() {
+        return hp <= maxHp * fleeThreshold;
```

Comment:
`tick` will be executed probably every frame. We could cache `maxHp * fleeThreshold` in a private property to not calculate it every time. But I'm not sure, it is very probable that either one of them could change on runtime?

Reply:
This is a good consideration. I am not sure if the monster maxHp ever changes once they are spawned.